src/containers/Headers.js

```diff
@@ -21,18 +21,47 @@ const mapDispatchToProps = dispatch => bindActionCreators({
 
 const hasClass = (className, classString) => classString.match(className) !== null
 
-const updateFilterSort = (id, isAsc, sorts) => [{id, isAsc}]
+const updateFilterSort = (id, isAsc) => [{id, isAsc}]
 
 const handlers = {
   updateSort: props => event => {
     const isAsc = !hasClass('sorted-asc', event.target.className)
     const id = event.target.getAttribute('data-id')
-    props.refreshData(props.name, {...props.filter, sorts: updateFilterSort(id, isAsc, props.filter.sorts)})
+    props.refreshData(props.name, {...props.filter, sorts: updateFilterSort(id, isAsc)})
+  }
+}
+
+const sort = props => {
+  // multi column sort support not implemented
+  // lowest sort integer column sorts asc
+  let sortRank
+  let sortId
+
+  const filterSort = def => {
+    if (!sortRank && def.sort) {
+      sortRank = def.sort
+      sortId = def.id
+    }
+    if (def.sort && def.sort < sortRank) {
+      sortId = def.id
+    }
+  }
+
+  props.definition.filter(filterSort)
+  if (sortId) {
+    props.refreshData(props.name, {...props.filter, sorts: updateFilterSort(sortId, true)})
+  }
+}
+
+const lifecycleHandlers = {
+  componentWillMount () {
+    sort(this.props)
   }
 }
 
 const enhance = compose(
   connect(mapStateToProps, mapDispatchToProps),
+  lifecycle(lifecycleHandlers),
```

Comment:
We shouldn't be triggering a new refresh for this, this logic should instead just update the initial state